### DISCUSSION TITLE
A mistake(?) in Std implementation

### DISCUSSION THREAD

**Discussion by EngineerKhan** [jax-ml/jax#8256] [Q&A] [answered]
I have made a simple snippet:

```
key = rand.PRNGKey(0)
x = rand.uniform(key, (1,100))
y = jnp.arange(1,101)

stdv = lambda x: jnp.linalg.norm(x-jnp.average(x))/(jnp.sqrt(len(x)))
```

When I call this `stdv` for both vectors and compare it with the JAX intrinsic one `jnp.std()`, it gives the same answers for `y`, but different for `x`. Why is it so?

**Comment by jakevdp** [comment, marked as answer]
Your `x` is 2-dimensional and `y` is one-dimensional. This means that `y.size` and `len(y)` are the same, but `x.size` and `len(x)` are not. If you switch `jnp.sqrt(len(x))` to `jnp.sqrt(x.size)`, your implementation will be correct for these inputs.
